ratingcurve/ratingmodel_builder.py

```diff
@@ -0,0 +1,503 @@
+"""Modifies PyMC ModelBuilder to work with the Rating model class"""
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+import numpy as np
+import pymc as pm
+import arviz as az
+import pandas as pd
+from pathlib import Path
+import json
+import math
+import warnings
+
+from .transform import LogZTransform
+from .model_builder import ModelBuilder
+
+if TYPE_CHECKING:
+    from arviz import InferenceData
+    from numpy.typing import ArrayLike
+    from pymc.util import RandomState
+    from typing import Any, Optional
+
+
+class RatingModelBuilder(ModelBuilder):
+    """Parent class for other rating models that sets not implemented
+    PyMC ModelBuilder class functions. Additionally, tweaks other ModelBuilder
+    functions for better application in rating curve fitting.
+    """
+
+    def __init__(self, **kwargs):
+        """Updates the ModelBuilder initialization to only configure the model.
+        Configuring the sampler now occurs in the `fit` call.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Additional keyword arguments to pass to the model configuration.
+        """
+        model_config = self.get_default_model_config(**kwargs)
+        self.model_config = model_config  # parameters for priors etc.
+
+        self.model = None  # Set by build_model
+        # idata is generated during fitting
+        self.idata: Optional[az.InferenceData] = None
+        self.is_fitted_ = False
+
+    def _data_setter(self,
```

Comment:
transform h and q here.
Use a unit transform for h. This may require you to update the normal prior for breakpoints.
If it becomes to complicated we can worry about h later.